src/Chunk.cpp

```diff
@@ -2076,36 +2074,73 @@ bool cChunk::DoWithEntityByID(UInt32 a_EntityID, cLambdaEntityCallback a_Callbac
 
 
 
-bool cChunk::ForEachBlockEntity(cBlockEntityCallback & a_Callback)
+// Helper function - there's probably a better place for this
+
+template <BLOCKTYPE, BLOCKTYPE...> bool IsOneOfImpl(BLOCKTYPE, std::false_type);
+template <class = void> bool IsOneOfImpl(BLOCKTYPE, std::true_type);
+
+template <BLOCKTYPE ... Types>
+bool IsOneOf(BLOCKTYPE x)
+{
+	return IsOneOfImpl<Types ...>(x,
+		std::integral_constant<bool, sizeof...(Types) == 0>()
+	);
+}
```

Comment:
Move the function to the top of the file (we "section" files based on globals / class A / class B / ...). Or even better, move it to src/BlockID.h (outside the `tolua_begin` .. `tolua_end` block).

Add a really good doxycomment about what this does, and all the edge cases.